cache.go

```diff
@@ -193,25 +193,28 @@ func (c *Cache[K, V]) get(key K, touch bool) *list.Element {
 // getWithOpts wraps the get method applying the given options.
 // Metrics are updated.
 // It returns nil if the item is not found or is expired.
-func (c *Cache[K, V]) getWithOpts(key K, opts ...Option[K, V]) *Item[K, V] {
+// The useLoader flag signals to use the loader if data not found in cache.
+// When using loader the cache will be unlocked, as the loader may take a long time.
+func (c *Cache[K, V]) getWithOpts(key K, useLoader bool, opts ...Option[K, V]) *Item[K, V] {
 	getOpts := options[K, V]{
 		loader:            c.options.loader,
 		disableTouchOnHit: c.options.disableTouchOnHit,
 	}
 
 	applyOptions(&getOpts, opts...)
 
-	c.items.mu.Lock()
 	elem := c.get(key, !getOpts.disableTouchOnHit)
-	c.items.mu.Unlock()
 
 	if elem == nil {
 		c.metricsMu.Lock()
 		c.metrics.Misses++
 		c.metricsMu.Unlock()
 
-		if getOpts.loader != nil {
-			return getOpts.loader.Load(c, key)
+		if useLoader {
+			c.items.mu.Unlock()
```

Comment:
Even though this is a working solution, it makes it hard to say what is exactly happening or why this mutex is needed. It also couples `getWithOpts()` with `Get()` because the mutex is locked there.

A clearer solution would be:
```go
func (c *Cache[K, V]) getWithOpts(key K, useCacheMutex, useLoader bool, opts ...Option[K, V]) *Item[K, V] {
    // ...
    if useCacheLock {
        c.items.mu.Lock()
    }
    
    elem := c.get(key, !getOpts.disableTouchOnHit)
    
    if useCacheLock {
        c.items.mu.Unlock()
    }
    // ...
}
```

Reply:
If you want, you can revert this back to what you had before and I'll fix this myself.

Reply:
Added the second flag @swithek

As you can see 2 flags are always true and false together.

Considering to merge them to 1.

Would you like propose a good name for the merge flag?

Reply:
You're right, we can merge them into `useLoader`, just be sure to document the mutex behaviour above `getWithOpts`.

I think after this this will be ready to be merged :)